tls_socket.js

```diff
@@ -506,11 +508,12 @@ exports.ensureDhparams = function (done) {
         return done(null, certsByHost['*'].dhparam);
     }
 
-    let filePath = tlss.cfg.main.dhparam;
-    if (!filePath) filePath = path.resolve(exports.config.root_path, 'dhparams.pem');
-    log.loginfo(`Generating a 2048 bit dhparams file at ${filePath}`);
+    const filePath = tlss.cfg.main.dhparam || 'dhparams.pem';
+    const fpResolved = path.resolve(exports.config.root_path, filePath);
 
-    const o = spawn('openssl', ['dhparam', '-out', `${filePath}`, '2048']);
+    log.loginfo(`Generating a 2048 bit dhparams file at ${fpResolved}`);
```

Comment:
Should be "Generating a 1024..." below you generate only a 1024 key...

Reply:
Good catch, I dropped that to sped it up during testing, I'll switch it back.